src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/WcfService.cs

```diff
@@ -424,6 +424,23 @@ public string EchoTimeAndSetCookie(string name)
             return value;
         }
 
+        public bool EchoHttpKeepAliveEnabled()
+        {
+            HttpRequestMessageProperty property;
+            object obj;
+            MessageProperties properties = new MessageProperties(OperationContext.Current.IncomingMessageProperties);
+
+            if (properties.TryGetValue(HttpRequestMessageProperty.Name, out obj))
+            {
+                property = obj as HttpRequestMessageProperty;
+                WebHeaderCollection collection = property.Headers;
+                string connectionValue = collection.Get(Enum.GetName(typeof(HttpRequestHeader), HttpRequestHeader.Connection));
+                return connectionValue.Equals("Close", StringComparison.OrdinalIgnoreCase);
+            }
```

Comment:
Doesn't this return true if the Connection header has the value `Close` meaning KeepAlive is disabled? Also this method doesn't echo anything back so having Echo in the name doesn't make sense. Can you either rename to `IsKeepAliveEnabled` and flip the logic or rename to `IsKeepAliveDisabled`. You are also returning true if there is a missing message property. Instead throw an exception as in this case we can't tell. This will be converted to a FaultException on the client so will signal something has gone wrong with the test scenario.